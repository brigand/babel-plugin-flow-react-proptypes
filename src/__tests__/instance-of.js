const babel = require('babel-core');
const content = `
import SomeClass from './some_class';
class SomeOtherClass {

}

class MyComponent extends React.Component {
    props: {
      some_class: SomeClass,
      some_other_class: SomeOtherClass,
      key_can_be_omitted?: SomeClass,
      value_can_be_null: ?SomeOtherClass,
    }
}
`;

it('instance-of', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['env', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
