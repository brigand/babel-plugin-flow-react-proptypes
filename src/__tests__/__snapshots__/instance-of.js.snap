// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`instance-of 1`] = `
"'use strict';

var _some_class = require('./some_class');

var _some_class2 = _interopRequireDefault(_some_class);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError(\\"this hasn't been initialised - super() hasn't been called\\"); } return call && (typeof call === \\"object\\" || typeof call === \\"function\\") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== \\"function\\" && superClass !== null) { throw new TypeError(\\"Super expression must either be null or a function, not \\" + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\\"Cannot call a class as a function\\"); } }

var SomeOtherClass = function SomeOtherClass() {
  _classCallCheck(this, SomeOtherClass);
};

var MyComponent = function (_React$Component) {
  _inherits(MyComponent, _React$Component);

  function MyComponent() {
    _classCallCheck(this, MyComponent);

    return _possibleConstructorReturn(this, (MyComponent.__proto__ || Object.getPrototypeOf(MyComponent)).apply(this, arguments));
  }

  return MyComponent;
}(React.Component);

MyComponent.propTypes = {
  some_class: typeof _some_class2.default === 'function' ? require('prop-types').instanceOf(_some_class2.default).isRequired : require('prop-types').any.isRequired,
  some_other_class: typeof SomeOtherClass === 'function' ? require('prop-types').instanceOf(SomeOtherClass).isRequired : require('prop-types').any.isRequired,
  key_can_be_omitted: typeof _some_class2.default === 'function' ? require('prop-types').instanceOf(_some_class2.default) : require('prop-types').any,
  value_can_be_null: typeof SomeOtherClass === 'function' ? require('prop-types').instanceOf(SomeOtherClass) : require('prop-types').any
};"
`;
